langs/cpp/cpp.c

```diff
@@ -0,0 +1,79 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+const char* clang = "/usr/bin/clang";
+const char* code = "/tmp/code.cpp";
+const char* object = "/tmp/code.o";
+const char* bin = "/tmp/code";
+
+int main(int argc, char* argv[]) {
+    if (argc > 1 && strcmp(argv[1], "--version") == 0) {
+        execv(clang, argv);
+        perror("execv");
+        return 0;
+    }
+
+    FILE* fp = fopen(code, "w");
+
+    if (!fp)
+        return 1;
+
+    char buffer[4096];
+    ssize_t nbytes;
+
+    while ((nbytes = fread(buffer, sizeof(char), sizeof(buffer), stdin)) > 0)
+        if (fwrite(buffer, sizeof(char), nbytes, fp) != nbytes)
+            return 2;
+
+    fclose(fp);
+
+    pid_t pid = fork();
+    if (!pid) {
+        // See https://clang.llvm.org/cxx_status.html for valid -std values.
+        execl(clang, clang, "-std=c++2b", "-target", "x86_64-alpine-linux-musl", "-O2", "-I/usr/include/c++/10.3.1/", "-I/usr/include/c++/10.3.1/x86_64-alpine-linux-musl/", "-I/usr/include/c++/10.3.1/backward/", "-o", object, "-c", code, NULL);
```

Comment:
Any reason to split the compile and link into two steps? It seems to me if you remove the `-c` and replace `object` with `bin` it should produce the executable directly.

Reply:
Seems like a good idea to simplify it a little bit.

Reply:
The way I did it was just the first thing that I thought of.

Reply:
Also maybe want to add `-fcolor-diagnostics` for colored error messages.